### ISSUE TITLE
Probably a very minor request, but can PECL's timezoneDB be made into a layer?

### ISSUE THREAD

**Issue by rquadling** [brefphp/extra-php-extensions#534]
The TimeZoneDB extension is pretty important when working with datetime on a global basis. The extension itself is VERY VERY minimal from a "code" perspective. It is more about "data" that is maintained externally and pulled into the extension when the things change.

It is a drop-in replacement to the version that is statically compiled into the PHP binary (assuming standard building) and is how all `date` functions, class, methods, etc. work. Add the drop-in, nothing except data changes and is only updated to be more accurate.

The updates to Olson and then the PECL TimeZoneDB are not related to deployments of PHP. From memory, the extension is compatible with ALL versions of PHP (it really is JUST some data ... a LOT of data though!).

**Comment by fredericgboutin-yapla** [comment]
Yeah, I agree, it is mind boggling how this is not a default extension in Docker official images as well because it leads to stupid bugs with timezone.

**Comment by mnapoli** [comment]
Hey! I wonder if this is (or could be) available through a Composer package? Distribution and installation would be much simpler.

**Comment by fredericgboutin-yapla** [comment]
I wondered the very same thing this morning 😅 but I doubt. When I think about the `DateTime` object and nest/carbon, they all seem to rely on the PHP framework itself.

It seems there is nothing of interest - https://packagist.org/?query=timezone

To be fair, the extension documentation stipulates this,

> You should only install this extension in case you need to get a later version of the timezone database than the one that ships with PHP - https://pecl.php.net/package/timezonedb

The more I think about this the more I think this is related to the active support state of the PHP version in use.

For example, PHP 7.4, which is not supported anymore, has a database version `2022.1` on `7.4.33` which released on Nov 2022.

![image](https://github.com/user-attachments/assets/684350fa-5707-45c6-9767-9e0738ee54d8)

https://onlinephp.io/c/d7311

I guess, what I'm trying to say is that, I think that as long as people are using an up to date / supported version of PHP, then the timezone database is fairly up to date already.

**Comment by mnapoli** [comment]
> I guess, what I'm trying to say is that, I think that as long as people are using an up to date / supported version of PHP, then the timezone database seems fairly up to date already.

oh good to know!

**Comment by rquadling** [comment]
You can't always update to the latest release as there may be dependencies that are not compatible. As this specific extension is pretty much universal, being able to update it would be a "good thing".

As PHP is NOT responsible for countries around the world changing, adding historic, or correcting their timezone information, and those changes NOT coinciding with an official PHP release for this 1 extension, the extension is very "different" in this regard. It is pretty much just data. Being able to add it in an easy manner (ideally on change from PECL), or, at least, a well documented process that doesn't get in the way of BrefPHP doing its thing, would be ideal.
